src/users/serializers.py

```diff
@@ -58,3 +58,14 @@ class Meta:
         fields = ('id', 'profile', 'active', 'major', 'bio', 'gpa', 'clubs', 'classes', 'pros', 'cons',)
         read_only_fields = ('id',)
 
+
+    def update(self, instance, validated_data):
+        if 'major' in validated_data:
+            major_obj = validated_data.pop('major')
+            major_text = major_obj['name']
+            major_query = Major.objects.filter(name=major_text)
+        if major_query.exists():
```

Comment:
doesnt this line throw `major_query` not defined if `'major' not in validated_data`